### ISSUE TITLE
create plugin for tinymce in canvas

### ISSUE THREAD

**Issue by cvong** [instructure/canvas-lms#378]
Hi Sir,

I would like to add a plugin (ICE) to tinymce in canvas. I wonder if there is any documentation for doing this. Thanks.

Chi-Hou

**Comment by claydiffrient** [comment]
@cvong Sorry for the delayed response here.  You've probably figured it out by now. I can't seem to find any documentation about how to do this, however I think that looking at `/app/coffeescripts/tinymce.coffee` would be a great place to start.  This may let you just declare the plugin there and it get loaded with TinyMCE whenever it loads.

**Comment by Aetherus** [comment]
I'm struggling to add tinymce plugins too. So far I've found that the source code involving tinymce scatters in the directories `app/coffeescripts/views/tinymce/`, `public/javascripts/tinymce_plugins/` and `app/views/jst/tinymce/`. The implementation of a plugin involves `RequireJS`, `Backbone`, `jQuery`, `handlebar` and God knows how many other stuff. What I want to do is just add a tab that contains only a simple `select` that enables the user to pick a video from the files he/she has uploaded (with a twisted URL). I found myself getting lost in such a whole bunch of mess.
